velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -331,9 +335,12 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
     } else {
       if (schemaElement.repetition_type ==
           thrift::FieldRepetitionType::REPEATED) {
-        VELOX_CHECK_LE(
-            children.size(), 2, "children size should not be larger than 2");
-        if (children.size() == 1) {
+        if (schema[parentSchemaIdx].converted_type ==
+            thrift::ConvertedType::LIST) {
```

Comment:
Can we add the `VELOX_CHECK_EQ(children.size(), 1)` here and `VELOX_CHECK_EQ(children.size(), 2)` in the map case